src/wasm/wasm-emscripten.cpp

```diff
@@ -921,11 +992,23 @@ void AsmConstWalker::addImports() {
     }
     wasm.table.initial.addr += queuedTableEntries.size();
   }
+
+  for (const auto& entry : globalImports) {
+    auto* import = new Global;
+    import->name = entry.second;
+    import->module = "GOT.func";
+    import->base = entry.first;
+    import->type = i32;
+    import->mutable_ = true;
+    wasm.addGlobal(import);
+  }
+
+  for (const auto& name : gotTable) {
+    wasm.removeGlobal(name);
```

Comment:
Wait then we remove the GOTs that we started with?

Oh this is because we start by importing em_asm with a bunch of signatures, but after we do our processing we collapse that down to just a few. That's nonobvious. I want to say "a comment might help!" but I can't think of what to say or where to put it.

Can we remove the gotTable before we add the globalImports though? I don't think there would be any name conflicts, but if it's the other way round I don't have to think about that case at all.

Reply:
Seems like we can remove the `gotTable` before we add the imports. This change has been made. Hopefully the added comments will make it clear what's happening.